tests/Proto.Actor.Tests/EventStreamTests.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Collections.Generic;
+using Xunit;
+using Microsoft.Extensions.Logging;
+
+namespace Proto.Tests
+{
+    public class EventStreamTests
+    {
+        [Fact]
+        public void EventStream_CanSubscribeToSpecificEventTypes()
+        {
+            var received = "";
+            var eventStream = new EventStream();
+            eventStream.Subscribe<string>(theString => received = theString);
+            eventStream.Publish("hello");
+            Assert.Equal("hello", received);
+        }
+
+        [Fact]
+        public void EventStream_CanSubscribeToAllEventTypes()
+        { 
+            var receivedEvents = new List<object>();
+            var eventStream = new EventStream();
+            eventStream.Subscribe(@event => receivedEvents.Add(@event));
+            eventStream.Publish("hello");
+            eventStream.Publish(1);
+            eventStream.Publish(true);
+            Assert.Equal(3, receivedEvents.Count);
+        }
+
+        [Fact]
+        public void EventStream_CanUnsubscribeFromEvents()
+        {
+            var receivedEvents = new List<object>();
+            var eventStream = new EventStream();
+            var subscription = eventStream.Subscribe<string>(@event => receivedEvents.Add(@event));
+            eventStream.Publish("first message");
+            subscription.Unsubscribe();
+            eventStream.Publish("second message");
+            Assert.Equal(1, receivedEvents.Count);
+        }
+
+        [Fact]
+        public void EventStream_OnlyReceiveSubscribedToEventTypes()
+        {
+            var eventsReceived = new List<object>();
+            var eventStream = new EventStream();
+            eventStream.Subscribe<int>(@event => eventsReceived.Add(@event));
+            eventStream.Publish("not an int");
+            Assert.Equal(0, eventsReceived.Count);
+        }
+
+        [Fact]
+        public void EventStream_DeadLetterEventsAreLogged()
+        {
+            var logger = new MockLogger();
+            var pid = new PID("test", "test");
+            var eventStream = new EventStream(logger);
+            eventStream.Publish(new DeadLetterEvent(pid, "dead", null));
+            eventStream.Publish("not a dead letter event");
+            Assert.Equal(1, logger.Events.Count);
+        }
+
+        private class MockLogger : ILogger {
```

Comment:
Is mocking and testing the logs a good idea?  I've always found it to be a pain.